lib/utils/resolveSilent.cjs

```diff
@@ -0,0 +1,50 @@
+// NOTICE: This file is generated by Rollup. To modify it,
+// please instead edit the ESM counterpart and rebuild with Rollup (npm run build).
+'use strict';
+
+const fs = require('node:fs');
+const path = require('node:path');
+const node_url = require('node:url');
+const importMetaResolve = require('@dual-bundle/import-meta-resolve');
+const resolveFrom = require('resolve-from');
+
+const pathSuffixes = ['', '.js', '.json', `${path.sep}index.js`, `${path.sep}index.json`];
+
+const specifierSuffixes = ['', '.js', '.json', '/index.js', '/index.json'];
+
+/**
+ * @param {string} parent
+ * @param {string} lookup
+ * @return {string | undefined}
+ */
+function resolveSilent(parent, lookup) {
+	if (path.isAbsolute(lookup)) {
+		for (const suffix of pathSuffixes) {
+			const filename = lookup + suffix;
+
+			if (fs.existsSync(filename)) {
+				return filename;
+			}
+		}
+
+		return;
+	}
+
+	const resolvedParent = node_url.pathToFileURL(path.resolve(parent, 'noop.js'));
```

Comment:
[ask] Why is `noop.js` necessary?

Reply:
To mock `import.meta.url` which is a filename.

Reply:
Is the mock necessary? Is a parent directory not working? In my PoC below, the parent directory seems to work: 🤔 

`tmp/a.js`:

```js
import { pathToFileURL } from 'node:url';

import { resolve } from '@dual-bundle/import-meta-resolve';

const parent = pathToFileURL(process.env.NODE_PATH);

console.log('parent:', parent.href);
console.log('import.meta.resolve:', import.meta.resolve('npm', parent));
console.log('import-meta-resolve:', resolve('npm', parent));
```

Run:

```sh-session
$ NODE_PATH=$(npm root -g) node --experimental-import-meta-resolve tmp/a.mjs
parent: file:///Users/masafumi.koba/.nvm/versions/node/v21.6.1/lib/node_modules
import.meta.resolve: file:///Users/masafumi.koba/.nvm/versions/node/v21.6.1/lib/node_modules/npm/index.js
import-meta-resolve: file:///Users/masafumi.koba/.nvm/versions/node/v21.6.1/lib/node_modules/npm/index.js

$ node -v
v21.6.1
```

Reply:
Make sense. Let's keep it as-is. 👍🏼